constantine/proof_systems/constraint_systems/r1cs_circom_parser.nim

```diff
@@ -20,15 +20,20 @@ import
   ../../serialization/[io_limbs, parsing],
   ../../platforms/[fileio, abstractions]
 
+# We use `sortedByIt` to sort the different sections in the file by their
+# `R1csSectionKind`
+from std / algorithm import sortedByIt
+
 type
-  R1csSectionKind = enum
-    kHeader                 = 1
-    kConstraints            = 2
-    kWire2LabelId           = 3
-    kCustomGatesList        = 4
-    kGustomGatesApplication = 5
+  R1csSectionKind = enum        # Actual values according to spec in comments. To use it as discriminator in  variant object
+    kHeader                 = 0 # 1
```

Comment:
Is it necessary to not use the spec value? Otherwise we can create a kPlaceholder for value 0 if we want an enum without hole.

Reply:
We can indeed just add a placeholder.